AttitudeManager/ControlAlgorithms/Inc/AM_ControlInterface.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * AM_ControlInterface.hpp
+ *
+ * Attitude Manager Control Algorithm Interface
+ *
+ * Created on: Oct 22, 2022
+ * Author(s): Aidan Bowers
+ */
+#ifndef ZPSW3_AM_CONTROL_INTERFACE_HPP
+#define ZPSW3_AM_CONTROL_INTERFACE_HPP
+
+#include "AM_DataTypes.hpp"
+
+namespace AM {
+
+class ControlInterface {
+   public:
+    ControlInterface(uint8_t num_actuators) : num_actuators(num_actuators){};
+
+    /// @brief Run the controls algorithm for the given flight model.
+    ///
+    /// @param instructions
+    ///   The instructions for the flight model to follow
+    /// @param outputs
+    ///   An array of actuator percentage throttles that the flight model
+    ///   believes should be used before mixing to meet the instructions
+    /// @param outputs_length
+    ///   The length of the outputs array. This should be equivilent to the
+    ///   number of actuator channels
+    virtual std::vector<ActuatorOutput> runControlsAlgo(
+        const AttitudeManagerInput &instructions) const = 0;
```

Comment:
I'm sure you already know but vectors use dynamic memory allocation, if you don't see this as a major issue or this is literally the only to do it, so be it!

Reply:
I was avoiding dynamic memory allocation before this, but was much much cleaner and easier to work with. I don't think the size of this will be a big concern to be honest. The vector will never be bigger than the number of actuators on the aircraft.